### ISSUE TITLE
Socket FD leak when disconnect from Android emulator

### ISSUE THREAD

**Issue by dogtopus** [frida/frida-core#284]
Initiate a session with `dev = frida.get_{usb_,}device` -> `app = dev.spawn` -> `session = dev.attach(app)` -> `script = session.create_script` -> `script.load` -> `dev.resume(app)` chain and then tear it down later with `script.unload` -> `session.detach` chain results in socket FD leak visible on lsof. Also looks liek there's no documented API to properly close the device.

Python code to reproduce the issue (similar behavior can also be observed on the nodejs binding so I assume that it's a core issue):
```python
import frida
import sys

if __name__ == '__main__':
    dev = None
    session = None
    script = None
    try:
        dev = frida.get_device('emulator-5554')
        app = dev.spawn(['com.android.settings'])
        session = dev.attach(app)
        script = session.create_script('console.log("Hello")')
        script.on('message', lambda msg: print(msg))
        script.load()
        dev.resume(app)
        # lsof here shows that 3 TCP sockets to ADBD are opened, which should be normal.
        _ = input('Frida instance started, press Enter to teardown...')
    finally:
        if script is not None:
            script.unload()
            script = None
        if session is not None:
            session.detach()
            session = None
        # no close() for dev?
        dev = None
        # lsof here shows that 2 TCP sockets to ADBD are still opened. Resource leak!
        _ = input('Frida instance stopped, press Enter to quit...')
        sys.exit(0)
```

**Comment by oleavr** [comment]
Thanks for reporting! That's not a leak though.

1. The backend itself keeps the ADB connection alive so it can notify you about devices being added/removed. I.e. through the `DeviceManager` signals.
2. The connection to the remote `frida-server` is opened on demand and then kept open. If it drops it's going to throw away the FD. And on the next API call on the `Device` it's going to attempt to reestablish it.
